app/src/main/java/com/github/se/polyfit/viewmodel/meal/MealViewModel.kt

```diff
@@ -30,30 +32,70 @@ constructor(private val user: User, private val mealRepo: MealFirebaseRepository
   val isComplete: LiveData<Boolean> = _isComplete
 
   init {
-    if (firebaseID.isNotEmpty()) {
-      mealRepo.getMeal(firebaseID).addOnCompleteListener {
-        if (it.isSuccessful) {
-          _meal.value = it.result
-          _isComplete.value = it.result?.isComplete() ?: false
+
+    viewModelScope.launch {
+      if (firebaseID.isNotEmpty()) {
+        mealRepo.getMeal(firebaseID).let {
+          if (it != null) {
+            _meal.value = it
+            _isComplete.value = it.isComplete()
+          }
         }
+      } else {
+        _meal.value = initialMeal?.copy() ?: Meal.default()
+        _isComplete.value = _meal.value?.isComplete() ?: false
       }
-    } else {
-      _meal.value = initialMeal?.copy() ?: Meal.default()
-      _isComplete.value = _meal.value?.isComplete() ?: false
+
+      _meal.observeForever { initialMeal = it }
     }
+  }
 
-    _meal.observeForever { initialMeal = it }
+  fun setNewMealObserver(liveMeal: LiveData<Meal>) {
+    _meal.removeObserver {}
+
+    liveMeal.observeForever {
+      _meal.value = it
+      _isComplete.value = it.isComplete()
+    }
   }
 
   fun setMealData(meal: Meal) {
     _meal.value = meal
+    _isComplete.value = meal.isComplete()
+  }
+
+  /** Allows for setting individual meal data fields instead of setting the whole meal */
+  fun setMealData(
+      mealOccasion: MealOccasion = _meal.value?.occasion ?: Meal.default().occasion,
+      name: String = _meal.value?.name ?: Meal.default().name,
+      mealID: Long = _meal.value?.mealID ?: Meal.default().mealID,
+      mealTemp: Double = _meal.value?.mealTemp ?: Meal.default().mealTemp,
+      ingredients: MutableList<Ingredient> = _meal.value?.ingredients ?: Meal.default().ingredients,
+      nutritionalInformation: NutritionalInformation =
+          _meal.value?.nutritionalInformation ?: Meal.default().nutritionalInformation,
+      firebaseID: String = _meal.value?.firebaseId ?: Meal.default().firebaseId,
+      createdAt: LocalDate = _meal.value?.createdAt ?: Meal.default().createdAt
+  ) {
+    _meal.value =
+        Meal(
+            mealOccasion,
+            name,
+            mealID,
+            mealTemp,
+            nutritionalInformation,
+            ingredients,
+            firebaseID,
+            createdAt)
+
+    _isComplete.value = _meal.value?.isComplete() ?: false
   }
 
   fun setMealName(name: String) {
```

Comment:
Now that we have `setMealData`, `setMealName` is not really needed, we should maybe remove it and just use `setMealData` for all name-changing place